src/test/storage/encoded_segment_test.cpp

```diff
@@ -303,4 +303,187 @@ TEST_F(EncodedSegmentTest, SegmentReencoding) {
   EXPECT_SEGMENT_EQ_ORDERED(value_segment, encoded_segment);
 }
 
+// Testing the internal data structures of Run Length-encoded segments for monotonically increasing values
+TEST_F(EncodedSegmentTest, RunLengthEncodingMonotonicallyIncreasing) {
+  constexpr auto row_count = int32_t{100};
+  auto values = pmr_concurrent_vector<int32_t>(row_count);
+
+  for (auto row_id = int32_t{0u}; row_id < row_count; ++row_id) {
+    values[row_id] = row_id;
+  }
+
+  const auto value_segment = std::make_shared<ValueSegment<int32_t>>(std::move(values));
+  const auto encoded_segment =
+      this->encode_segment(value_segment, DataType::Int, SegmentEncodingSpec{EncodingType::RunLength});
+
+  const auto run_length_segment = std::dynamic_pointer_cast<const RunLengthSegment<int32_t>>(encoded_segment);
+  ASSERT_TRUE(run_length_segment);
+  EXPECT_EQ(run_length_segment->values()->size(), row_count);
+  EXPECT_EQ(run_length_segment->end_positions()->size(), row_count);
+  EXPECT_EQ(run_length_segment->end_positions()->front(), 0ul);
+  EXPECT_EQ(run_length_segment->end_positions()->back(), 99ul);
+}
+
+TEST_F(EncodedSegmentTest, RunLengthEncodingVaryingRuns) {
+  constexpr auto row_count = int32_t{100};
+  auto values = pmr_concurrent_vector<int32_t>(row_count);
+
+  // fill first ten values with decreasing values
+  for (auto row_id = int32_t{0u}; row_id < 10; ++row_id) {
+    values[row_id] = 10 - row_id;
+  }
+
+  // the next 8 values are repeated 10 times each (i.e., 80 values)
+  auto row_id = 10;
+  for (auto value = int32_t{0}; value < 8; ++value) {
+    for (auto repetition = 0; repetition < 10; ++repetition) {
+      values[row_id] = value;
+      ++row_id;
+    }
+  }
+
+  // fill remainder with increasing values
+  while (row_id < row_count) {
+    values[row_id] = row_id;
+    ++row_id;
+  }
+
+  const auto value_segment = std::make_shared<ValueSegment<int32_t>>(std::move(values));
+  const auto encoded_segment =
+      this->encode_segment(value_segment, DataType::Int, SegmentEncodingSpec{EncodingType::RunLength});
+
+  const auto run_length_segment = std::dynamic_pointer_cast<const RunLengthSegment<int32_t>>(encoded_segment);
+
+  ASSERT_TRUE(run_length_segment);
+  EXPECT_EQ(run_length_segment->values()->size(), 10 + 8 + 10);
+  EXPECT_EQ(run_length_segment->end_positions()->size(), 10 + 8 + 10);
+  EXPECT_EQ(run_length_segment->end_positions()->front(), 0ul);
+  EXPECT_EQ(run_length_segment->end_positions()->back(), 99ul);
+
+  // first longer run has the value 10, starts at position 10, and ends at position 19
+  EXPECT_EQ(run_length_segment->values()->at(10), 0);
+  EXPECT_EQ(run_length_segment->end_positions()->at(9), 9);
+  EXPECT_EQ(run_length_segment->end_positions()->at(10), 19);
```

Comment:
Just like above, sth. like `9 + num_repetitions` could be clearer. Anyways, I do not know if that's necessary to understand the test.